### PR TITLE
Add SpotBugs,fix npe findings. Test travis build fails with spotted bugs

### PR DESCRIPTION
**Pull request by banjaxxer**
Signed-off-by: Steve Bolton <boltonst@uk.ibm.com>

Please don't merge this if it passes the build as I am waiting for delivery of common, maven and gradle new versions first.  
Plus I want to look at the spotbugs findings that are left
But I want to test my goals with Travis and ensure the build fails with spotbugs:check when there are findings.

### CODE REVIEW

**Comment by banjaxxer** [comment]
I modified the build script to add verify goals before spotbugs - especially for the deploy goal I want to be sure the build verify runs before spotbugs and deploy only runs after.  I don't know if I need to do this or if the goals will sort themselves out.

**Comment by banjaxxer** [comment]
Spotbugs exclusion filter works

File: .travis/script.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@ set -x
 
 # If this is a master build then deploy at the end
 if [ "$TRAVIS_BRANCH" = 'master' ] && [ "$TRAVIS_PULL_REQUEST" == 'false' ]; then
-  mvn -Dstyle.color=always -B -U -P sign --settings travis-settings.xml deploy;
```
Comment (stewartfrancis) [inline .travis/script.sh:6]: You don't need to add verify here, deploy is after verify in the maven build lifecycle

File: pom.xml [hunk 2]
```diff
@@ -50,7 +50,37 @@
         <groupId>org.sonatype.plugins</groupId>
         <artifactId>nexus-staging-maven-plugin</artifactId>
       </plugin>
+      <plugin>
```
Comment (stewartfrancis) [inline pom.xml:53]: You've also moved the nexus-staging-maven-plugin from pluginManagement to plugins, it needs to be moved back

File: src/main/java/com/ibm/cics/bundle/parts/BundlePublisher.java [hunk 3]
```diff
@@ -111,28 +100,35 @@ public void addResource(BundleResource resource) throws PublishException {
...
-		String extension = fileName.substring(dot + 1).toLowerCase();
-		BundlePartType type = BundlePartType.getType(extension);
-		if(type != null) {
-			String name = fileName.substring(0, dot);
-			if (name.length() > 0) {
-				return new Define(name, type, toImport.toString());
-			} else {
-				log.debug("Couldn't determine bundle part name for file \"" + fileName + "\"");
-				return null;
+		if (toImport != null) {
```
Comment (stewartfrancis) [inline src/main/java/com/ibm/cics/bundle/parts/BundlePublisher.java:103]: Presumably the right thing to do in this situation is Objects.requireNonNull(toImport)?  A null parameter to this method isn't meaningful, and throwing an NPE is probably the right thing to do (which would've happened anyway I suppose).

File: pom.xml (hunk 2 above)
Comment (banjaxxer) [reply pom.xml:53]: Sad face

File: src/main/java/com/ibm/cics/bundle/parts/BundlePublisher.java (hunk 3 above)
Comment (banjaxxer) [reply src/main/java/com/ibm/cics/bundle/parts/BundlePublisher.java:103]: Fixed

File: pom.xml [hunk 4]
```diff
@@ -109,6 +124,10 @@
             <failOnWarnings>true</failOnWarnings>
           </configuration>
         </plugin>
+        <plugin>
```
Comment (stewartfrancis) [inline pom.xml:127]: You still have changes to nexus-staging-maven-plugin's config.  I'm not expecting to see any

**Pull request merged**